pkg/controller/core/workload_controller.go

```diff
@@ -252,3 +266,27 @@ func workloadFinished(w *kueue.Workload) bool {
 	}
 	return false
 }
+
+// We do not verify Pod's RuntimeClass legality here as this will be performed in admission controller.
+func handlePodOverhead(wl *kueue.Workload, classes nodev1.RuntimeClassList) {
+	for _, pod := range wl.Spec.PodSets {
+		if pod.Spec.RuntimeClassName != nil && len(pod.Spec.Overhead) == 0 {
+			for _, item := range classes.Items {
```

Comment:
It might be faster to just do a Get for each podset (in general there would be 1 or 2 podsets)

Reply:
make sense.